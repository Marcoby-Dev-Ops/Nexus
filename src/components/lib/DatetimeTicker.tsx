import React from 'react';

/**
 * @name DatetimeTicker
 * @description Displays the current date and time, updating every second.
 * @returns {JSX.Element}
 */
const DatetimeTicker: React.FC = () => {
  const [now, setNow] = React.useState(new Date());

  React.useEffect(() => {
    const timer = setInterval(() => setNow(new Date()), 1000);
    return () => clearInterval(timer);
  }, []);

  return (
    <div className="flex flex-col items-end mr-2 select-none" aria-live="polite">
      <span className="text-xs text-muted-foreground font-medium">
        {now.toLocaleDateString(undefined, { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' })}
      </span>
      <span className="text-lg font-mono tabular-nums text-primary">
        {now.toLocaleTimeString(undefined, { hour: '2-digit', minute: '2-digit', second: '2-digit' })}
      </span>
    </div>
  );
};

export default DatetimeTicker; 