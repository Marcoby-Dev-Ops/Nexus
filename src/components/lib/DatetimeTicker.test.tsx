/**
 * @file DatetimeTicker.test.tsx
 * @description Unit and snapshot tests for the DatetimeTicker component.
 */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import DatetimeTicker from '../lib/DatetimeTicker';

describe('DatetimeTicker', () => {
  it('renders without crashing', () => {
    render(<DatetimeTicker />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<DatetimeTicker />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 