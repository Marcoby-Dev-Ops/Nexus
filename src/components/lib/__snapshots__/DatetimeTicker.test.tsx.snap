// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatetimeTicker matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="flex flex-col items-end mr-2 select-none"
  >
    <span
      class="text-xs text-muted-foreground font-medium"
    >
      Sunday, June 8, 2025
    </span>
    <span
      class="text-lg font-mono tabular-nums text-primary"
    >
      09:09:25 PM
    </span>
  </div>
</DocumentFragment>
`;
